src/Adapter/Currency/CommandHandler/EditCurrencyHandler.php

```diff
@@ -229,4 +247,30 @@ private function assertDefaultCurrencyIsNotBeingRemovedOrDisabledFromShop(Curren
             }
         }
     }
+
+    /**
+     * @param Currency $entity
+     * @param string $newIsoCode
+     */
+    private function updateNameAndSymbol(Currency $entity, $newIsoCode)
+    {
+        $contextLocale = Context::getContext()->language->getLocale();
```

Comment:
Inject the context into the constructor, will be easier to maintain

Reply:
Ok, but I don't see much value in it, this won't be unit testable anyway.

Reply:
Everything is unit testatble  :smile: 
Anyway, if we use singleton everywhere, it will be really hard to migrate to on a new "context" in the future.
Just keep in mind, I need data from another class, I inject it.

Reply:
I don't see how hiding context (in this case in yml service declaration) would make it easier to refactor, maybe even harder, as the context dependency is not explicit.

Reply:
When I said, context, I mean here the Language object or something similar to retrieve data

Reply:
It's still the same, even if I inject `Language` it can be considered as injecting context. If we were to follow object oriented design, then language should come via method argument instead imo.